src/util/misc/projectStroke.ts

```diff
@@ -39,62 +54,287 @@ export const projectStrokeOnPoints = (
     strokeUniform,
     scaleX,
     scaleY,
+    skewX,
+    skewY
   }: projectStrokeOnPointsOptions,
   openPath: boolean
-): Point[] => {
-  const coords: Point[] = [],
+): returnedProjections => {
+
+  const coords: returnedProjections = [],
     s = strokeWidth / 2,
+    scale = new Point(scaleX, scaleY),
     strokeUniformScalar = strokeUniform
       ? new Point(1 / scaleX, 1 / scaleY)
       : new Point(1, 1),
-    getStrokeHatVector = (v: Point) => {
-      const scalar = s / Math.hypot(v.x, v.y);
-      return new Point(
-        v.x * scalar * strokeUniformScalar.x,
-        v.y * scalar * strokeUniformScalar.y
-      );
+    scaleHatVector = (hatVector: Point, scalar: number) => {
+      return hatVector.multiply(strokeUniformScalar).scalarMultiply(scalar);
+    },
+    applySkew = (v: Point) => {
+      let vector = new Point(v);
+      vector.y += vector.x * Math.tan(degreesToRadians(skewY)); // skewY must be applied before skewX as this distortion affects skewX calculation
+      vector.x += vector.y * Math.tan(degreesToRadians(skewX));
+      return vector;
     };
+
   if (points.length <= 1) {
     return coords;
   }
+
+  /*
+    Functions to find projections for each type of line join (or open path)
+    MDN: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linejoin
+    Spec: https://svgwg.org/svg2-draft/painting.html#StrokeLinejoinProperty
+    Playground to understand how the line joins works: https://hypertolosana.github.io/efficient-webgl-stroking/index.html
+    View the calculated projections for each of the control points: https://codesandbox.io/s/project-stroke-points-with-context-to-trace-b8jc4j?file=/src/index.js
+  */
+
+  /**
+   * OPEN PATH START/END - Line cap: Butt
+   * If open path, no matter the type of the line join, the line is always the same
+   * Calculation: to find the projections, just find the points orthogonal to that stroke 
+   * 
+   * @see https://github.com/fabricjs/fabric.js/pull/8344#1-1-butt
+   */
+  const projectionsOpenPathButt = (
```

Comment:
I was avoiding this because it's too much context to pass through the functions. I made the changes ([#034a90a](https://github.com/fabricjs/fabric.js/pull/8344/commits/034a90a4de872246ac283522d54bf4b6ea5a3a2b)), but it seems to me that it became less readable

Doing the transformations before passing doesn't seem appropriate to me, because the calculation logic will be split. For example, not all cases use the vectors `AB` and `AC` (see the case of the `projectionsRoundNoSkew` function).

Reply:
I will take a look
Can you run prettier so I can work on the code as well?

Reply:
@ShaMan123, just confirming, I commit only the prettier modified files that we're working on here (polyline.class.ts, projectStroke.ts, etc), right? Because it changed a bunch of files..

Reply:
Hmm.. prettier should have only changed the committed files.
The rest should be prettified already.
The files show up in github desktop but once you commit they are filtered out as empty.

Reply:
https://github.com/fabricjs/fabric.js/pull/8344/commits/1636dd98c20d972f0484425710afcec8d39e3fd5